### DISCUSSION TITLE
Question about addition of an extra Potential in the HF calculation

### DISCUSSION THREAD

**Discussion by mb6-31ansatz** [pyscf/pyscf#2652] [Q&A]
Hi, 
currently, I am working on a project that requires me to add a potential (constant initially) in the Hartree-Fock Hamiltonian. This means, that, I would like to have an Hamiltonian be composed by the sum of Core potential, Electronic Repulsion potential and My new Potential. 
How can I add this feature safely into the code?

PS At the moment, I do not plan to add this feature to PySCF. I am planning to use it locally, since it would be a very niche application.

**Comment by tberkel** [comment]
You can overwrite the `get_hcore()` function of the `mf` object to also include your custom potential, which must be expressed as a matrix of your potential in the AO basis. For an example with a completely custom Hamiltonian, see [here](https://github.com/pyscf/pyscf/blob/master/examples/scf/40-customizing_hamiltonian.py).

**Comment by verena-neufeld** [comment]
Hi @mb6-31ansatz , thanks for your question. Has it been answered and can this issue be closed?
